### PR TITLE
Make cmd an array and add a debug possibility

### PR DESCRIPTION
**Pull request by perlpunk**
* more readable
* easier to add optional args with cmd+=(...)
* avoid eval

### CODE REVIEW

File: script/openqa-clone-custom-git-refspec [hunk 1]
```diff
@@ -82,18 +82,30 @@ clone_job() {
...
+        "$host" "$job"
+        _GROUP="$GROUP"
+        TEST="$test"
+        BUILD="$build"
+        CASEDIR="$casedir"
+        PRODUCTDIR="$productdir"
+        NEEDLES_DIR="$needles_dir"
+        ${args[*]}
+    )
+    $debug && echo "cmd: ${cmd[@]}"
```
Comment (okurz) [inline script/openqa-clone-custom-git-refspec]: that would be the same as "dry_run", wouldn't it?

Comment (perlpunk) [reply script/openqa-clone-custom-git-refspec]: yeah, similar. I can take it out

Comment (kalikiana) [reply script/openqa-clone-custom-git-refspec]: How should `debug` and `dry_run` differ? Just the `cmd: ` prefix?

Neither is documented... perhaps adding docs would be the first step, the second would be seeing what's worth adding?

**Comment by okurz** [comment]
any plans to continue on this?

**Pull request closed without merging**